javascript-carousel/main.js

```diff
@@ -0,0 +1,110 @@
+const images = document.querySelectorAll('img');
+let count = 0;
+const dots = document.querySelectorAll('.fa-circle');
+const dot1 = document.querySelector('.dot1');
+const dot2 = document.querySelector('.dot2');
+const dot3 = document.querySelector('.dot3');
+const dot4 = document.querySelector('.dot4');
+const dot5 = document.querySelector('.dot5');
+
+// Auto carousel
+setInterval(function () {
+  images[count].className = 'hidden';
+  dots[count].classList.replace('fa-solid', 'fa-regular');
+  count++;
+  if (count === 5) {
+    count = 0;
+  }
+  images[count].className = '';
+  dots[count].classList.replace('fa-regular', 'fa-solid');
+}, 3000);
+
+// Dot progress bar
+dot1.addEventListener('click', function (event) {
+  if (!event.target.matches('.dot1')) {
+    return;
+  }
+  images[count].className = 'hidden';
+  dots[count].classList.replace('fa-solid', 'fa-regular');
+  if (dot1 === event.target) {
+    count = 0;
+  }
+  images[count].className = '';
+  dots[count].classList.replace('fa-regular', 'fa-solid');
+});
```

Comment:
The `event.target` checks aren't necessary here, since the event listener is attached directly to the target element.